### DISCUSSION TITLE
Material UI TextField Component of type="file" is giving "C:\fakepath\file.csv" instead of FileList

### DISCUSSION THREAD

**Discussion by saurabhpabhare** [react-hook-form/react-hook-form#4576] [Q&A] [answered]
I have been trying to use RHF with Material UI. So far I am very happy with RHF, but today I am stuck with file upload.

I am creating a form with a file field which accepts multiple files. 

Following is the link for my CSB -> [https://codesandbox.io/s/dazzling-driscoll-ksgnd?file=/src/index.js](url)

As you can you when i do a submit, log in console shows fakepath for the uploaded file instead of filelist.

I've tried passing inputRef={register} to my controller but that throws following error on submit.

`Failed to set the 'value' property on 'HTMLInputElement': This input element accepts a filename, which may only be programmatically set to the empty string.`

<img width="1792" alt="Screenshot 2021-03-31 at 11 51 09 PM" src="https://user-images.githubusercontent.com/30390186/113192121-04428380-927c-11eb-8d8e-7174c4b65d0a.png">

I have been at this for a long while now, so any pointers would be really great help. 

Thank you very much!

**Comment by bluebill1049** [comment, marked as answer]
can you try with `render` prop instead of `as` and assign those value and onChange prop individual, see if this will resolve the problem.

**Reply by saurabhpabhare** [reply]
yeah that works. Thank you so much. 😊
